tests/Integration/Behaviour/Features/Scenario/Cart/Adding/CartRule/usage_limit_per_user.feature

```diff
@@ -0,0 +1,29 @@
+# ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml --tags remove-cart-rule
+@reset-database-before-feature
+@cart-rule-usage-limit
+
+Feature: A cart rule's usage limit per user is detected
+
+  Background:
+    Given country "US" is enabled
+    And there is customer "testCustomer" with email "pub@prestashop.com"
+    And customer "testCustomer" has address in "US" country
+
+  Scenario: Cart rule usage limit is detected when the cart rule is created anonymously then assigned to a customer who already used it
+    Given I create an empty cart "dummy_cart" for customer "testCustomer"
+    And shop configuration for "PS_CART_RULE_FEATURE_ACTIVE" is set to 1
+    And there is a cart rule named "limitedCartRule" that applies an amount discount of 1.0 with priority 1, quantity of 2 and quantity per user 1
+    And cart rule "limitedCartRule" has a discount code "foo1"
+    And there is a product in the catalog named "product1" with a price of 19.812 and 1000 items in stock
```

Comment:
I think this part could be in the background In case we need to add other scenarios later